src/test/java/org/jilt/test/PropagatedAnnotationsTest.java

```diff
@@ -0,0 +1,47 @@
+package org.jilt.test;
+
+import org.jilt.test.data.annotations.NonParameterAnnotation;
+import org.jilt.test.data.annotations.PropagatedAnnotationNameBuilder;
+import org.jilt.test.data.annotations.PropagatedAnnotationNameBuilders;
+import org.junit.Test;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.reflect.Method;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PropagatedAnnotationsTest {
+    @Test
+    public void setter_in_builder_propagates_annotations() throws Exception {
+        Method firstNameSetter = PropagatedAnnotationNameBuilder.class.getMethod("firstName", String.class);
+
+        assertThat(firstNameSetter.getParameters()[0].getDeclaredAnnotation(CheckForNull.class)).isNotNull();
+
+        Method middleNameSetter = PropagatedAnnotationNameBuilder.class.getMethod("middleName", String.class);
+
+        assertThat(middleNameSetter.getParameters()[0].getDeclaredAnnotation(Nullable.class)).isNotNull();
+
+        Method lastNameSetter = PropagatedAnnotationNameBuilder.class.getMethod("lastName", String.class);
+
+        assertThat(lastNameSetter.getParameters()[0].getDeclaredAnnotation(Nonnull.class)).isNotNull();
+        assertThat(lastNameSetter.getParameters()[0].getDeclaredAnnotation(NonParameterAnnotation.class)).isNull();
```

Comment:
Let's group these together - this is pretty difficult to read right now:

```suggestion
        Method firstNameSetter = PropagatedAnnotationNameBuilder.class.getMethod("firstName", String.class);
        assertThat(firstNameSetter.getParameters()[0].getDeclaredAnnotation(CheckForNull.class)).isNotNull();

        Method middleNameSetter = PropagatedAnnotationNameBuilder.class.getMethod("middleName", String.class);
        assertThat(middleNameSetter.getParameters()[0].getDeclaredAnnotation(Nullable.class)).isNotNull();

        Method lastNameSetter = PropagatedAnnotationNameBuilder.class.getMethod("lastName", String.class);
        assertThat(lastNameSetter.getParameters()[0].getDeclaredAnnotation(Nonnull.class)).isNotNull();
        assertThat(lastNameSetter.getParameters()[0].getDeclaredAnnotation(NonParameterAnnotation.class)).isNull();
```